src/Resolvers/Microsoft.DotNet.NativeWrapper/NETEnvironmentInfo.cs

```diff
@@ -0,0 +1,107 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Runtime.InteropServices;
+using Microsoft.Deployment.DotNet.Releases;
+using static Microsoft.DotNet.NativeWrapper.Interop;
+
+namespace Microsoft.DotNet.NativeWrapper
+{
+    public interface INetBundleInfo
+    {
+        public ReleaseVersion Version { get; }
+
+        public string Path { get; }
+    }
+
+    public sealed class NetSdkInfo : INetBundleInfo
+    {
+        public ReleaseVersion Version { get; private set; }
+
+        public string Path { get; private set; }
+
+        public NetSdkInfo(string version, string path)
+        {
+            Version = new ReleaseVersion(version);
+            Path = path;
+        }
+    }
+
+    public sealed class NetRuntimeInfo : INetBundleInfo
+    {
+        public ReleaseVersion Version { get; private set; }
+
+        public string Path { get; private set; }
+
+        public string Name { get; private set; }
+
+        public NetRuntimeInfo(string name, string version, string path)
+        {
+            Version = new ReleaseVersion(version);
+            Path = path;
+            Name = name;
+        }
+    }
+
+    public sealed class NetEnvironmentInfo
+    {
+        public IEnumerable<NetRuntimeInfo> RuntimeInfo { get; private set; }
+
+        public IEnumerable<NetSdkInfo> SdkInfo { get; private set; }
+
+        public NetEnvironmentInfo(IEnumerable<NetRuntimeInfo> runtimeInfo, IEnumerable<NetSdkInfo> sdkInfo)
+        {
+            RuntimeInfo = runtimeInfo;
+            SdkInfo = sdkInfo;
+        }
+
+        public NetEnvironmentInfo()
+        {
+            RuntimeInfo = new List<NetRuntimeInfo>();
+            SdkInfo = new List<NetSdkInfo>();
+        }
+
+        internal void InitializeWindows(IntPtr info, IntPtr resultContext)
```

Comment:
Same comment as in the DllImports - why do we have to duplicate these? The code is effectively identical on all platforms.

Reply:
Also - at the very least we should compile only the version which is needed on a given platform. There's no need to ship Unix code on Windows and vice-versa.